### PR TITLE
Fixed wrong implementation of shouldPerformExposureDetection

### PR DESCRIPTION
**Pull request by inf2381**
## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description

Fixed the wrong implementation that lead to a too frequent check of exposures

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift [hunk 1]
```diff
@@ -31,6 +31,24 @@ struct RiskProvidingConfiguration {
...
+enum NextExposureDetection: Equatable {
+	case date(Date)
+	case now
+
+	static func == (lhs: NextExposureDetection, rhs: NextExposureDetection) -> Bool {
+		switch (lhs, rhs) {
+		case (.now, .now):
+			return true
+		case let (.date(a), .date(b)):
+			// return true if dates are less than 30 seconds apart
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift:44]: why 30 seconds? (having a comparison threshold makes perfect sense to me, I am just wondering if there is any further reasoning behind the value)

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift [hunk 2]
```diff
@@ -40,13 +58,13 @@ extension RiskProvidingConfiguration {
...
 
-	func nextExposureDetectionDate(lastExposureDetectionDate: Date?, currentDate: Date = Date()) -> Date {
+	func nextExposureDetectionDate(lastExposureDetectionDate: Date?, currentDate: Date = Date()) -> NextExposureDetection {
 		let potentialDate = Calendar.current.date(
 			byAdding: exposureDetectionInterval,
 			to: lastExposureDetectionDate ?? .distantPast,
 			wrappingComponents: false
 			) ?? .distantPast
-		return potentialDate > currentDate ? currentDate : potentialDate
+		return potentialDate > currentDate ? .date(potentialDate) : .now
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift:67]: 🥇

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/__tests__/RiskProvidingConfigurationTests.swift [hunk 3]
```diff
@@ -61,20 +61,29 @@ final class RiskProvidingConfigurationTests: XCTestCase {
...
-		let nextExpectedDate = Calendar.current.date(byAdding: detectionInterval, to: .distantPast)
 		let now = Date()
-		XCTAssertEqual(config.nextExposureDetectionDate(lastExposureDetectionDate: nil, currentDate: now), nextExpectedDate)
+		XCTAssertEqual(config.nextExposureDetectionDate(lastExposureDetectionDate: nil, currentDate: now), NextExposureDetection.now)
 	}
 
-	func testGetNextExposureDetectionDate_FutureLastDetectionDate() {
+	func testGetNextExposureDetectionDate_in22Hours() {
 		// Test the case where the last exposure detection date is in the future.
 		// This edge case should be handled by just returning now as the next detection date
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/__tests__/RiskProvidingConfigurationTests.swift:70]: is this comment still up-to-date?

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift (hunk 1 above)
Comment (johannesrohwer) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift:44]: -> marc told me its just an arbitrary value. fine for me!

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/__tests__/RiskProvidingConfigurationTests.swift (hunk 3 above)
Comment (johannesrohwer) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/__tests__/RiskProvidingConfigurationTests.swift:70]: I will take this over in my branch.

**Review by johannesrohwer — APPROVED** [review]

**Review by ChristianKienle — APPROVED** [review]
